app/controllers/profiles_controller.rb

```diff
@@ -1,5 +1,5 @@
 class ProfilesController < ApplicationController
   def show
-    @user = User.find(params[:id])
+    @profiled_user = Profile.username(params[:id])
```

Comment:
Since you asked, instance variables can lead to hard to spot bugs when re-using templates or splitting out templates into partials and re-using those elsewhere. These variables default to `nil` when they are undefined, so a template re-used elsewhere might render an empty string where maybe it wasn't intended to be empty. The alternative is to use a helper method, which will raise a `NoMethodError` when the template is used in a place where the method isn't defined.  So, in this particular instance, I would have something like this:

```ruby
def show
end

private

def profile
  @_profile ||= Profile.find_by(username: params[:id])
end
helper_method :profile
```

This exposes a `profile` method in the view, which will invoke this method when called. Does that help?

Reply:
Yes!

Reply:
Yes, that helps explain how `profile`  works as the helper_method - thank you - but I'm confused by the use of `find_by`. I thought `find_by` would only be used when the class inherits from ActiveRecord.

Reply:
speaking for ZP here, his example looks like what you'd typically see in `ProfilesController` where `Profile` is an AR backed model. for this application, I'm guessing that line meant to be
```ruby
@_profile ||= Profile.username(params[:id])
```

Reply:
FWIW, you can write a `find_by` method, but I have heard the argument that having one on a non-AR model can lead to future confusion and assumptions being made about the PORO that will obviously not be true. Personally, I prefer a common DSL amongst all Ruby objects that interact with the database in some fashion, whether they inherit from ActiveRecord or not. YMMV.

The intention of `Profile` here was to be a presenter object to sit in between the controller and the view. This provides some structure to the view while leaving the controller for responding to HTTP requests.  The alternative is to use the `User` class which may or may not need modified according to requirements of this view.

Reply:
Cool, thanks @zporter. As per our discussion, I'm going to leave the Profile class out for now (as I've already deleted it in a recent commit on this branch). However, once I get to modeling Votes, I may bring it back to serve as a presenter.